packit/upstream.py

```diff
@@ -856,3 +858,25 @@ def _get_rpms_from_rpmbuild_output(self, output: str) -> List[str]:
             )
 
         return rpms
+
+    def _expand_git_ref(self, ref: Optional[str]) -> str:
+        if not ref or not re.match(r".*[\[\?\*].*", ref):
+            logger.debug("No ref given or is not glob pattern")
+            return ref
```

Comment:
Hmm, escaping in regexes... What is the exact reason for that?

Reply:
- Can we have some comment about that?
- What about branch name as an `upstream-ref`? Will this work fine?

Reply:
> - Can we have some comment about that?

added todo list to the description ;)

> - What about branch name as an upstream-ref? Will this work fine?

Yes! :tada: I'm looking into it :) I'll push commit that will allow branches soonish